src/app/routes/onDemandTV/getInitialData/index.js

```diff
@@ -1,90 +1,94 @@
-import path from 'ramda/src/path';
+import pathOr from 'ramda/src/pathOr';
 import fetchPageData from '../../utils/fetchPageData';
 import overrideRendererOnTest from '../../utils/overrideRendererOnTest';
 import getPlaceholderImageUrl from '../../utils/getPlaceholderImageUrl';
+import pathWithLogging, {
+  LOG_LEVELS,
+} from '#lib/utilities/logging/pathWithLogging';
+import { TV_MISSING_FIELD, TV_EPISODE_EXPIRED } from '#lib/logger.const';
 
-const getBrandTitle = path(['metadata', 'title']);
-const getLanguage = path(['metadata', 'language']);
-const getHeadline = path(['promo', 'headlines', 'headline']);
-const getId = path(['metadata', 'id']);
-const getShortSynopsis = path(['promo', 'media', 'synopses', 'short']);
-const getMasterBrand = path(['metadata', 'createdBy']);
-const getContentType = path(['metadata', 'analyticsLabels', 'contentType']);
-const getPageTitle = path(['metadata', 'analyticsLabels', 'pageTitle']);
-const getPageIdentifier = path([
-  'metadata',
-  'analyticsLabels',
-  'pageIdentifier',
-]);
-const getEpisodeId = path(['content', 'blocks', 0, 'id']);
-const getReleaseDateTimeStamp = path(['metadata', 'releaseDateTimeStamp']);
-const getDurationISO8601 = path([
-  'promo',
-  'media',
-  'versions',
-  0,
-  'durationISO8601',
-]);
-const getThumbnailImageUrl = json =>
-  getPlaceholderImageUrl(path(['promo', 'media', 'imageUrl'], json));
-const getPromoBrandTitle = path(['promo', 'brand', 'title']);
-const getImageUrl = path(['content', 'blocks', 0, 'imageUrl']);
-const getEpisodeAvailableFrom = path([
-  'content',
-  'blocks',
-  '0',
-  'versions',
-  '0',
-  'availableFrom',
-]);
-const getEpisodeAvailableUntil = path([
-  'content',
-  'blocks',
-  '0',
-  'versions',
-  '0',
-  'availableUntil',
-]);
+import nodeLogger from '#lib/logger.node';
 
-const getEpisodeAvailability = ({ availableFrom, availableUntil }) => {
+const logger = nodeLogger(__filename);
+
+const getEpisodeAvailability = ({ availableFrom, availableUntil, url }) => {
   const timeNow = Date.now();
-  if (!availableUntil || timeNow < availableFrom) return false;
+  if (!availableUntil || timeNow < availableFrom) {
+    logger.info(TV_EPISODE_EXPIRED, {
```

Comment:
Ideally we would follow the single responsibility principle here, and not combine get episode availability with logging.

Would it be possible to look at what was done for onDemandRadio https://github.com/bbc/simorgh/blob/latest/src/app/routes/onDemandRadio/getInitialData/index.js, and consider refactoring for max reusability?

Reply:
Most of this function is combining getting and logging, as well as the function itself (`**_get_**InitialData`)

There's an alternative here: https://github.com/bbc/simorgh/compare/od-tv-logging...od-tv-logging-a but I think it's an unnecessary complexity

Reply:
Keen to get more opinions on this

Reply:
The favoured approach on logging regarding the single responsibility principle is to delegate to a logger, which is what is implemented here. The function isn't handling implementation of logging, it's passing that responsibility over to the available logger. I'd say good to go.

Reply:
It rustles me jimmies (Ryan, 2020) that we have different implementations of the same logic for OD-radio & OD-tv here. `getUrl` as well as calling the logger within `getEpisodeAvailability` are essentially replicating what's being done within OD-radio's `logInitialData` file, right?

Maybe this should all be pushed into a `logEpisodeUnavailable` logging util akin to `pathWithLogging` that would then be used by both radio & TV (removing od-radio's `logInitialData`)?

Reply:
Alright - I'll check with Jane if the expired episode for TV and Radio is the same (including this issue: https://github.com/bbc/simorgh/issues/6889)

If so, I'll move this function into the `lib` folder after https://github.com/bbc/simorgh/pull/6970 is merged